test/mpi/rma/lock_nested.c

```diff
@@ -0,0 +1,66 @@
+/* -*- Mode: C; c-basic-offset:4 ; -*- */
+/*
+ * (C) 2016 by Argonne National Laboratory.
+ *     See COPYRIGHT in top-level directory.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <mpi.h>
+
+#define BUFSIZE 4
+
+/* This program tests nested lock. Process 0 locks the other processes
+ * one by one, then unlock each of them.*/
+
+#define ERR_REPORT_EXIT(str,...) do {                                \
+        fprintf(stderr, str, ## __VA_ARGS__);fflush(stderr);    \
+        MPI_Abort(MPI_COMM_WORLD, 1);                           \
+    } while (0);
+
+int main(int argc, char *argv[])
+{
+    int rank = 0, nprocs = 0, dst = 0;
+    int mpi_errno = MPI_SUCCESS;
+    int winbuf[BUFSIZE];
+    MPI_Win win = MPI_WIN_NULL;
+
+    MPI_Init(&argc, &argv);
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &nprocs);
+
+    memset(winbuf, 0, sizeof(int) * BUFSIZE);
+    MPI_Win_create(winbuf, sizeof(int) * BUFSIZE, sizeof(int), MPI_INFO_NULL, MPI_COMM_WORLD, &win);
+
+    if (rank == 0) {
+        /* lock each process */
+        for (dst = 0; dst < nprocs; dst++) {
+            mpi_errno = MPI_Win_lock(MPI_LOCK_SHARED, dst, 0, win);
+            if (mpi_errno != MPI_SUCCESS) {
```

Comment:
FATAL(abort) is fine, just in case some MPI implementations do not abort for the lock error, so I need to check the return value as well.